packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManager.java

```diff
@@ -129,6 +133,28 @@ protected T prepareToRecycleView(@NonNull ThemedReactContext reactContext, T vie
     return view;
   }
 
+  @Override
+  public void onLayoutChange(View v,
+                             int left,
+                             int top,
+                             int right,
+                             int bottom,
+                             int oldLeft,
+                             int oldTop,
+                             int oldRight,
+                             int oldBottom) {
+    // Old width and height
+    int oldWidth = oldRight - oldLeft;
+    int oldHeight = oldBottom - oldTop;
+
+    // Current width and height
+    int currentWidth = right - left;
+    int currentHeight = bottom - top;
+    if ((currentHeight != oldHeight || currentWidth != oldWidth) && this.transformOrigin != null && this.transform != null) {
+      setTransformProperty((T) v, this.transform, this.transformOrigin);
+    }
+  }
```

Comment:
Instead of using `onLayoutChange` and adding a layout listener, can you override `layout` (which RN calls to apply the layout)